types.ts

```diff
@@ -0,0 +1,36 @@
+import * as t from "io-ts";
+
+export const ProtocolIo = t.type({
```

Comment:
sick. never used io-ts but have used similar libs and they are really excellent for this exact kinna thing 🔥 🔥 🔥